internal/controller/dnsrecord_healthchecks_test.go

```diff
@@ -0,0 +1,181 @@
+//go:build integration
+
+package controller
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	. "github.com/onsi/gomega/gstruct"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/kuadrant/dns-operator/api/v1alpha1"
+	"github.com/kuadrant/dns-operator/pkg/builder"
+)
+
+var _ = Describe("DNSRecordReconciler_HealthChecks", func() {
+	var (
+		dnsRecord *v1alpha1.DNSRecord
+
+		testNamespace, testHostname string
+	)
+
+	BeforeEach(func() {
+		CreateNamespace(&testNamespace)
+
+		testZoneDomainName := strings.Join([]string{GenerateName(), "example.com"}, ".")
+		testHostname = strings.Join([]string{"foo", testZoneDomainName}, ".")
+
+		dnsProviderSecret := builder.NewProviderBuilder("inmemory-credentials", testNamespace).
+			For(v1alpha1.SecretTypeKuadrantInmemory).
+			WithZonesInitialisedFor(testZoneDomainName).
+			Build()
+		Expect(k8sClient.Create(ctx, dnsProviderSecret)).To(Succeed())
+
+		dnsRecord = &v1alpha1.DNSRecord{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      testHostname,
+				Namespace: testNamespace,
+			},
+			Spec: v1alpha1.DNSRecordSpec{
+				RootHost: testHostname,
+				ProviderRef: v1alpha1.ProviderRef{
+					Name: dnsProviderSecret.Name,
+				},
+				Endpoints:   getTestLBEndpoints(testHostname),
+				HealthCheck: getTestHealthCheckSpec(),
+			},
+		}
+	})
+
+	It("Should create valid probe CRs and remove them on DNSRecord deletion", func() {
```

Comment:
Creating a trivial integration check. At this moment we can only verify the creation of CR to our liking and a correct removal. Will expand in tests as the work progresses